src/lib.rs

```diff
@@ -98,14 +114,72 @@ pub use util::ecdsa::PrivateKey;
 #[deprecated(since = "0.26.1", note = "Please use `ecdsa::PublicKey` instead")]
 pub use util::ecdsa::PublicKey;
 
+#[cfg(feature = "std")]
 use std::io;
+#[cfg(not(feature = "std"))]
+use core2::io;
+
+#[cfg(not(feature = "std"))]
+mod io_extras {
+    /// A writer which will move data into the void.
+    pub struct Sink {
+        _priv: (),
+    }
+
+    /// Creates an instance of a writer which will successfully consume all data.
+    pub const fn sink() -> Sink {
+        Sink { _priv: () }
+    }
+
+    impl core2::io::Write for Sink {
+        #[inline]
+        fn write(&mut self, buf: &[u8]) -> core2::io::Result<usize> {
+            Ok(buf.len())
+        }
+
+        #[inline]
+        fn flush(&mut self) -> core2::io::Result<()> {
+            Ok(())
+        }
+    }
+}
+
+mod prelude {
+    #[allow(unused_imports)]
+    #[cfg(all(not(feature = "std"), not(test)))]
+    pub use alloc::{string::{String, ToString}, vec::Vec, boxed::Box, borrow::{Cow, ToOwned}, slice, rc, sync};
+
+    #[allow(unused_imports)]
+    #[cfg(any(feature = "std", test))]
+    pub use std::{string::{String, ToString}, vec::Vec, boxed::Box, borrow::{Cow, ToOwned}, slice, rc, sync};
+
+    #[allow(unused_imports)]
+    #[cfg(all(not(feature = "std"), not(test)))]
+    pub use alloc::collections::{BTreeMap, BTreeSet, btree_map};
```

Comment:
Why is this different from the std version? Things seem to build warning free for me without the unused_imports allows (though because its a pub-use I dont think rustc actually *checks* anything).

Reply:
I think it's because the serde feature is broken with no-std, 
@devrandom can you fix no-std+serde?(it shouldn't be too hard I think)

Reply:
@TheBlueMatt - are you asking about BTreeSet?  not sure why it was in there - removed.

@elichai no-std + serde is fixed (as well as some other combinations)

also removed the unused_imports, they were leftovers